src/lib/network_pool/transacation_pool_reorg_tests.ml

```diff
@@ -0,0 +1,433 @@
+    | Zkapp_blocking_send {fee; _} -> fee
+  
+end
+
+(** Main generator for prefix, minor and major sequences. This generator has a more firm grip
+     on how data is generated than usual. It uses Command_spec and Account_spec modules for 
+     user command definitions which then are carved into Signed_command list. By default generator
+     fulfill standard use cases for ledger reorg, like merging transactions from minor and major sequences
+     with preference for major sequence as well as 2 additional corner cases:
+
+     ### Edge Case : Nonce Precedence
+
+      - In major sequence, transactions update the account state to a point where the nonce of the account is smaller 
+         than the first nonce in the sequence of removed transactions.
+      - The mempool logic determines that if this condition is true, the entire minor sequence should be dropped.
+
+     ### Edge Case : Nonce Intersection
+
+      - Transactions using the same account appear in all three sequences (prefix, minor, major)
+  
+    On top of that one can enable/disable two special corner cases (permission change and limited capacity)
+  *)
+let gen_branches init_ledger_state ~permission_change ~limited_capacity ?(sequence_max_length=3) () = 
+  let open Quickcheck.Generator.Let_syntax in
+  let%bind prefix_length = Int.gen_incl 1 sequence_max_length in
+  let%bind branch_length = Int.gen_incl 1 sequence_max_length in
+    
+  let spec = Array.map init_ledger_state ~f:Account_spec.of_ledger_row
+  in
+  
+  let%bind prefix_command_spec = 
+    Command_spec.gen_sequence spec ~length:prefix_length in
+  
+  let minor = Array.copy spec in
+  let%bind minor_command_spec = 
+    Command_spec.gen_sequence minor ~length:branch_length in
+  
+  let major = Array.copy spec in
+  let%bind major_command_spec = 
+    Command_spec.gen_sequence major ~length:branch_length in
+
+  (* Optional Edge Case 1: Limited Account Capacity
+
+    - In major sequence*, a transaction `T` from a specific account decreases its balance by amount `X`.
+    - In minor sequence*, the same account decreases its balance in a similar transaction `T'`, but by an amount much smaller than `X`, followed by several other transactions using the same account.
+    - The prefix ledger* contains just enough funds to process major sequence, with a small surplus.
+    - When applying *minor sequence* without the transaction `T'` (of the same nonce as the large-amount transaction `T` in major sequence), 
+        the sequence becomes partially applicable, forcing the mempool logic to drop some transactions at the end of *minor sequence*.
+  *)
+
+  (*find account in major and minor brnaches with the same nonces and similar balances (less than 100k mina diff)*)
+  let minor_acc_opt = Array.find_map minor ~f:(fun minor_acc -> 
+      Array.find major ~f:(fun major_acc -> 
+            Int.equal minor_acc.nonce major_acc.nonce &&
+            minor_acc.balance - major_acc.balance < 100_000_000_000
```

Comment:
What is the chance that such two accounts won't be found?

Won't the probability of the test being executed be too low?